### PR TITLE
Arm Combat Cybernetic Implants (because toolset arm shanks weren't enough)

### PR DESCRIPTION
**Pull request by silicons**
Adds a bunch of arm implants.
Energy blade projector, mounted medical beamgun, stun-arm implant (Well it's adminspawn so not a balance issue :^)), flash implant, and one implant with all four. Also, a surgical toolkit implant.
Flash implant doubles as a high-power flashlight when extended :D

:cl: Mekhi
add: Nanotrasen has been experimenting on cybernetics for a while, and rumors are they have a few combat prototypes available... 
experiment: New arm implants: Energy-blade projector, implanted medical beamgun, stun-arm implant (Like the borg version), flash implant that is automatically-regenerating and doubles as a powerful flashlight. There is also one with all four items in one. Also, a surgery toolkit implant. Adminspawn only for the moment.
/:cl:

I would add the esword implant to nukeops/maybe even traitor for a high, high price, and maybe flash implant to R&D, but I'd rather not a feature PR get delayed by balance issues..

### CODE REVIEW

**Comment by PKPenguin321** [comment]
I'd been meaning to add an arm mounted energy blade to nuke ops ever since I added the arm mod system. I just never got sprites

If this is just adminspawned stuff nix the change log. If not please state how it is acquired

**Comment by kevinz000** [comment]
@PKPenguin321  I'd rather add them to nukeops/other stuff (I mean, the flash implant isn't THAT OP compared to the others), but there's a lot of balance implications so I'd leave it in a separate PR.

File: code/modules/projectiles/guns/medbeam.dm [hunk 1]
```diff
@@ -108,8 +109,10 @@
 /obj/item/weapon/gun/medbeam/proc/on_beam_tick(var/mob/living/target)
 	if(target.health != target.maxHealth)
 		PoolOrNew(/obj/effect/overlay/temp/heal, list(get_turf(target), "#80F5FF"))
-	target.adjustBruteLoss(-4)
-	target.adjustFireLoss(-4)
+	target.adjustBruteLoss(-healpower)
+	target.adjustFireLoss(-healpower)
+	target.adjustToxLoss(-healpower)
+	target.adjustOxyLoss(-healpower)
```
Comment (X-TheDark) [inline code/modules/projectiles/guns/medbeam.dm]: Undocumented changes (ToxLoss/OxyLoss heal).

Comment (kevinz000) [inline code/modules/projectiles/guns/medbeam.dm]: ah

**Comment by Bawhoppen** [comment]
:-1: on the medigun changes

**Comment by kevinz000** [comment]
@Bawhoppen Why? They're medical NANITES. Why shouldn't they be able to fix tox and oxyloss when every other nanite chemical in the game (well, there's two others) does?

**Comment by Bawhoppen** [comment]
balance, and not everything has to do everything

**Comment by InsaneHyena** [comment]
I want robotics to be able to do something other than make mechs. This is great, and I've been waiting for this.

**Comment by KorPhaeron** [comment]
Put mediun shit in new pr

**Comment by kevinz000** [comment]
K

**Comment by kevinz000** [comment]
done

File: code/game/objects/items/robot/robot_items.dm [hunk 2]
```diff
@@ -10,15 +10,16 @@
...
+/obj/item/borg/stun/attack(mob/living/M, mob/living/user)
 	if(ishuman(M))
 		var/mob/living/carbon/human/H = M
 		if(H.check_shields(0, "[M]'s [name]", src, MELEE_ATTACK))
 			playsound(M, 'sound/weapons/Genhit.ogg', 50, 1)
 			return 0
-
-	if(!user.cell.use(charge_cost))
-		return
+	if(isrobot(user))
```
Comment (optimumtact) [inline code/game/objects/items/robot/robot_items.dm]: you copypasted this from stunprod code didn't you?

File: code/modules/surgery/organs/augments_arms.dm [hunk 3]
```diff
@@ -68,6 +68,13 @@
 		"<span class='notice'>[holder] snaps back into your [zone == "r_arm" ? "right" : "left"] arm.</span>",
 		"<span class='italics'>You hear a short mechanical noise.</span>")
 
+	if(istype(holder, /obj/item/device/assembly/flash/armimplant))
+		var/obj/item/device/assembly/flash/F = holder
+		F.SetLuminosity(0)
```
Comment (optimumtact) [inline code/modules/surgery/organs/augments_arms.dm]: why not set these by default on the subtype and then just update icon?

File: code/game/objects/items/robot/robot_items.dm (hunk 2 above)
Comment (kevinz000) [inline code/game/objects/items/robot/robot_items.dm]: i didnt even look at stunprod code but if you want this refactor'd/whatnot just say so

File: code/modules/surgery/organs/augments_arms.dm (hunk 3 above)
Comment (kevinz000) [inline code/modules/surgery/organs/augments_arms.dm]: uh :v no idea what that means.

Comment (kevinz000) [inline code/modules/surgery/organs/augments_arms.dm]: actually hold on

**Comment by kevinz000** [comment]
@Kromgar mebbe but there's going to be balance concerns with a knife that you can't take out of them :^)

**Comment by kevinz000** [comment]
Also, we already have integrated toolset implants, those have 15 burn weapons built into them and a 20 brute combat knife if you emag it.

**Comment by PKPenguin321** [comment]
2 more things
1. you still did not remove the changelog
2. do you actually have sprites for these because if not bleh

**Comment by kevinz000** [comment]
1. ...? What? Remove the changelog? Why?
2. I don't see the need for unique sprites for anything but the stunarm which I don't know what to give it? It's a stunarm.

File: code/modules/assembly/flash.dm [hunk 4]
```diff
@@ -190,3 +190,32 @@
...
+/obj/item/device/assembly/flash/armimplant/burn_out()
+	if(I)
+		I.owner << "<span class='warning'>Your photon projector implant overheats and deactivates!</span>"
+		I.Retract()
+	overheat = 1
+	addtimer(src, "cooldown", flashcd * 2)
+
+/obj/item/device/assembly/flash/armimplant/try_use_flash(mob/user = null)
+	if(overheat)
+		if(I)
```
Comment (optimumtact) [inline code/modules/assembly/flash.dm]: combine these two if checks into one

    if(I && I.owner)

Also, why do you not check I.owner exists in other procs, why just this one?

File: code/modules/assembly/flash.dm [hunk 5]
```diff
@@ -190,3 +190,32 @@
...
+			if(I.owner)
+				I.owner << "<span class='warning'>Your photon projector is running too hot to be used again so quickly!</span>"
+		return 0
```
Comment (optimumtact) [inline code/modules/assembly/flash.dm]: use TRUE/FALSE defines for when 0 and 1 represent boolean values please.

**Comment by PKPenguin321** [comment]
The need for unique sprites is so you can distinguish at a glance that it's undroppable etc
Same reason chainsaw has a different inhand from mounted chainsaw

**Pull request merged**